src/main/java/org/generationcp/middleware/api/germplasm/GermplasmServiceImpl.java

```diff
@@ -125,6 +133,312 @@ public GermplasmServiceImpl(final HibernateSessionProvider sessionProvider) {
+		}
+
+		// Retrieve location and method IDs in one go
+		final Map<String, Integer> locationAbbreviationIdMap = this.getLocationAbbreviationIdMap(germplasmUpdateDTOList);
+		final Map<String, Method> codeBreedingMethodDTOMap =
+			this.getCodeBreedingMethodDTOMap(germplasmUpdateDTOList);
+
+		// Retrieve the names and attributes associated to GIDs in one go.
+		final Map<Integer, List<Name>> namesMap =
+			this.daoFactory.getNameDao().getNamesByGidsInMap(germplasmList.stream().map(g -> g.getGid()).collect(Collectors.toList()));
+		final List<Attribute> attributes =
+			this.daoFactory.getAttributeDAO()
+				.getAttributeValuesGIDList(germplasmList.stream().map(g -> g.getGid()).collect(Collectors.toList()));
+		final Map<Integer, List<Attribute>> attributesMap =
+			attributes.stream().collect(Collectors.groupingBy(Attribute::getGermplasmId, LinkedHashMap::new, Collectors.toList()));
+
+		for (final Germplasm germplasm : germplasmList) {
+			this.saveGermplasmUpdateDTO(attributeCodesFieldNoMap, nameCodesFieldNoMap,
+				germplasmUpdateDTOMap,
+				locationAbbreviationIdMap, codeBreedingMethodDTOMap, namesMap, attributesMap, germplasm, conflictErrors);
+		}
+
+		if (!conflictErrors.isEmpty()) {
+			throw new MiddlewareRequestException(null, conflictErrors);
+		}
+		return germplasmList.stream().map(g -> g.getGid()).collect(Collectors.toSet());
+
+	}
+
+	private void saveGermplasmUpdateDTO(final Map<String, Integer> attributeCodes, final Map<String, Integer> nameCodes,
+		final Map<String, GermplasmUpdateDTO> germplasmUpdateDTOMap, final Map<String, Integer> locationAbbreviationIdMap,
+		final Map<String, Method> codeBreedingMethodDTOMap, final Map<Integer, List<Name>> namesMap,
+		final Map<Integer, List<Attribute>> attributesMap, final Germplasm germplasm, final Multimap<String, Object[]> conflictErrors) {
+		final Optional<GermplasmUpdateDTO> optionalGermplasmUpdateDTO =
+			this.getGermplasmUpdateDTOByGidOrUUID(germplasm, germplasmUpdateDTOMap);
+		if (optionalGermplasmUpdateDTO.isPresent()) {
+			final GermplasmUpdateDTO germplasmUpdateDTO = optionalGermplasmUpdateDTO.get();
+			final Optional<Method> breedingMethodDtoOptional =
+				Optional.ofNullable(codeBreedingMethodDTOMap.getOrDefault(germplasmUpdateDTO.getBreedingMethodAbbr(), null));
+			final Optional<Integer> locationIdOptional =
+				Optional.ofNullable(locationAbbreviationIdMap.getOrDefault(
+					StringUtils.isNotEmpty(germplasmUpdateDTO.getLocationAbbreviation()) ?
+						germplasmUpdateDTO.getLocationAbbreviation().toUpperCase() : StringUtils.EMPTY, null));
+			final Optional<Integer> germplasmDateOptional =
+				StringUtils.isEmpty(germplasmUpdateDTO.getCreationDate()) ? Optional.empty() :
+					Optional.ofNullable(Integer.parseInt(germplasmUpdateDTO.getCreationDate()));
+			final Optional<String> referenceOptional =
+				StringUtils.isEmpty(germplasmUpdateDTO.getReference()) ? Optional.empty() :
+					Optional.ofNullable(germplasmUpdateDTO.getReference());
```

Comment:
Should just be: `Optional.of(Integer.parseInt(germplasmUpdateDTO.getReference()))`